homeassistant/helpers/schema_config_entry_flow.py

```diff
@@ -188,7 +191,10 @@ def _show_next_step(
 
         # Show form for next step
         return self._handler.async_show_form(
-            step_id=next_step_id, data_schema=data_schema, errors=errors
+            step_id=next_step_id,
+            data_schema=data_schema,
+            errors=errors,
+            last_step=form_step.next_step is None,
```

Comment:
The `next_step` function is allowed to return `None`. Maybe add a comment to `SchemaFlowFormStep` mentioning that providing a `last_step` function will always set `last_step` to False because we don't know if the function will return `None` or not.

Another option would be to retype `SchemaFlowFormStep.next_step` and not allow it to return None, since it's not a functionality that is needed today.